### ISSUE TITLE
Problem with width of text label

### ISSUE THREAD

**Issue by dawidr** [openlayers/openlayers#10559]
I'm using OpenLayers v6.1.1 and for some reason, from time to time, after a few refreshes of the page, the background of the text label gets the wrong height.
It should look like this:
![image](https://user-images.githubusercontent.com/575077/72753536-783bc300-3bc5-11ea-9af7-b6f3f43a3604.png)
but finally, it will become:
![image](https://user-images.githubusercontent.com/575077/72753553-838eee80-3bc5-11ea-960b-e8b5885cf3b2.png)
which is wrong.

Here is a piece of code I'm using to generate this style:

```
style = new ol.style.Style({
	image: new ol.style.Icon({
		anchor: [0.5, 46],
		anchorXUnits: 'fraction',
		anchorYUnits: 'pixels',
		src: icon
	}),
});
if (SOME_CONDITION) {
	style.setText(new ol.style.Text({
		font: 'bold 11px Arial, Helvetica, sans-serif',
		fill : new ol.style.Fill({
			color : 'white',
		}),
		backgroundFill: new ol.style.Fill({
			color : highlighted ? 'rgba(250, 44, 47, 0.8)' : 'rgba(119, 118, 115, 0.8)'
		}),
		padding: [2, 2, 2, 2],
		text: name,
		offsetY: -5
	}));
}

layer = new ol.layer.Vector({
    source: my_source
    style: getFeatureStyle() //above code is used insode of this function to set the style for all features on this layer
});
```

Is this a bug or I'm doing something wrong?

**Comment by ahocevar** [comment]
I think this was a bug that we recently fixed. Can you try the latest dev Version (`npm install ol@dev`)?

**Comment by dawidr** [comment]
@ahocevar I'm not using NPM or Node.js, I'm including `ol.js` into my project. Can I grab the latest dev in *.js file?

**Comment by ahocevar** [comment]
@dawidr I just updated the master build, so you can now get the latest from https://openlayers.org/en/master/build/ol.js (js) and https://openlayers.org/en/master/css/ol.css (css).

**Comment by dawidr** [comment]
@ahocevar Unfortunate, the issue is not gone. I've used files which you mention but it looks even worse right now:

![2020-02-03_20-58-50](https://user-images.githubusercontent.com/575077/73686500-64658600-46c8-11ea-9494-1d4c0278ac4c.jpg)

**Comment by ahocevar** [comment]
@dawidr Can you please create an example that shows the issue on https://codesandbox.io/? Thanks!

**Comment by sc-dawidrutkowski** [comment]
I'm on the latest v6.2.1 and the issue is still there for me and I'm out of ideas.
@ahocevar You asked me about creating an example. That might be hard as our code is soooo complex and big that extracting code would be extremely hard.

A attached an example code how I'm adding features to the layer. There is no rocket science, everything looks fine. So I've no idea what might be wrong. Maybe it is related to the fact that we are moving features on the map like that:

```var feature = this.source.getFeatureById(device.DeviceId);
if (feature) {
    feature.setGeometry(new ol.geom.Point(ol.proj.transform([device.Longitude, device.Latitude], 'EPSG:4326', 'EPSG:3857')));
    feature.set("icon", device.IconName);
}
```

Can you give me some tips about where to look for that issue? You mention that it should be already fixed - what was the nature of that problem?

**Comment by ahocevar** [comment]
@sc-dawidrutkowski There were several changes recently regarding font rendering, also after 6.2.1. Changing the geometries to move feature should not be causing any issues.

You're saying it's no rocket science - so it should be possible to provide a codesandbox. All I need is an example that creates a vector layer with styles similar to the one your application uses. Just enough to reproduce the issue. So maybe don't try to extract code, but try to reproduce the bug from scratch.

**Comment by dawidr** [comment]
I was playing a bit with the styles and even if I remove most of the styles the problem is still there. So it is not related the images:

```
style = new ol.style.Style({
    // image: new ol.style.Icon({
    //     anchor: [0.5, 24],
    //     anchorXUnits: 'fraction',
    //     anchorYUnits: 'pixels',
    //     src: icon
    // }),
    text: new ol.style.Text({
        //font: 'bold 11px Arial, Helvetica, sans-serif',
        fill: new ol.style.Fill({
            color: 'white',
        }),
        backgroundFill: new ol.style.Fill({
            color: highlighted ? 'rgba(250, 44, 47, 0.8)' : 'rgba(119, 118, 115, 0.8)'
        }),
        //padding: [2, 2, 2, 2],
        text: deviceName,
        //offsetY: 17
    })
});
```

The background for the features has wrong width:

![image](https://user-images.githubusercontent.com/575077/77054402-0a96f200-69d0-11ea-829d-9d9557887064.png)

**Comment by ahocevar** [comment]
@dawidr Thanks for your continued research on this. Can you please try to put these findings into a standalone minimal example so we can debug the problem?

**Comment by dawidr** [comment]
@ahocevar I was trying to create minimalistic example but in all those cases it works fine. In the meantime, I found another place where it is not working. Different layer, different text label, different code and the same problem. Take a look:

![image](https://user-images.githubusercontent.com/575077/77833469-ce852f00-713d-11ea-862e-d5dd6f072d0d.png)

And it will back to normal after a few page refreshes:

![image](https://user-images.githubusercontent.com/575077/77833486-04c2ae80-713e-11ea-90ff-6745210311a4.png)

In my opinion, it is definitely something in the library. But I've no clear steps to reproduce...

**Comment by ahocevar** [comment]
@dawidr Sorry to hear you're still struggling with this, but without a minimal example we won't be able to hunt this bug.

**Comment by dawidr** [comment]
@ahocevar I will not be able to prepare minimal example as in minimal ones it works fine. I'm 100% that there is an issue with `backgroundFill` property of `ol.style.Text` so that I remove it and I'm using `stroke` now.

**Comment by Doskii** [comment]
I have the same issue except mine are even higher and all the time, not just sometimes. I'm basically using the exact same code on my website as I'm using on a minimalized test page but only the website has this issue, just like you mentioned.

![image](https://user-images.githubusercontent.com/36999268/114718362-36e98300-9d36-11eb-95e5-2ee541566e91.png)

I'm using OpenLayers 6.4.3 but both 6.5 and 5.1.3 seem to show the same problem.

Did you ever find the solution or did you just settle with using the stroke?

**Comment by dawidr** [comment]
@Doskii I was sure that this issue is not just on my side! I was not able to fix it... And because of that, I gave up, remove the background and move to stroke. As I've written in one of the previous comments: https://github.com/openlayers/openlayers/issues/10559#issuecomment-611132701.

**Comment by Doskii** [comment]
@dawidr Ok, so after some more headbanging against my desk I've noticed that if I remove my `display: flex` from the `body` element it fixes the problem (but breaks my site of-course). However if I put a `div` just below the `body` and give it `flex` the problem is gone. It seem to have something to do with body having flex. I don't understand why this would be a problem but oh well. I've reported it as a new issue #12213, though I don't know if this issue is related to yours or not as you mentioned it only happening sometimes 🤔 

In my case I can't just change the structure of the website whenever I feel like so I'll try to figure out another way. If you're interested this is basically what happened for me: https://jsfiddle.net/Doskii/29hxu4bd/
